en/contributing.md

```diff
@@ -0,0 +1,384 @@
+    controller can connect to the KVM host using a passphraseless private SSH
+    key.
+
+!!! Warning: 
+    Data will be lost unless you do the right thing.
+
+!!! Positive "High score":
+    A positive note that should include a title.
+
+!!! Negative "Game over": 
+    A negative note that should include a title.
+
+!!! Positive "": 
+    I'm done, and I feel fine.
+
+### Foldouts
+
+When a page contains a lot of extraneous information such as walkthroughs
+containing many images or reference tables, a *foldout* can be used. This will
+create a collapsed header which, when clicked, will expand to display all the
+content below it.
+
+```no-highlight
+^# Header
+  Content can be multi-paragraphed and will be sent through the Markdown parser
+
+  as long as content is continually indented under the header.
+```
+
+
+### Hyperlinks
+
+Links to internal files or external URLs use the following format:
+
+```no-highlight
+[visible text][label]
+```
+
+The `visible text` is what will appear on the web page. The `label` is used to
+refer to the destination, which is placed at the bottom of the file:
+
+```
+<!-- LINKS -->
+
+[label]: destination
+```
+
+For example:
+
+```no-highlight
+- For more on this topic see [DHCP][dhcp].
+- To understand haproxy, see the [upstream configuration manual][upstream-haproxy-manual].
+
+...
+
+[dhcp]: installconfig-networking-dhcp.md
+[upstream-haproxy-manual]: http://cbonte.github.io/haproxy-dconv/1.6/configuration.html
+```
+
+The visible text should use an active style as opposed to a passive style. For
+instance, try to avoid:
+
+```no-highlight
+A [proxy][maas-proxy] can optionally be configured.
+```
+
+Notes:
+
+- An internal page is referred to by its source filename.
+- Try to use the same label:destination pair throughout the documentation.
+
+
+### Images
+
+An image should not be overly cropped - allow for context. When ready, place
+the image file in the `media` directory.
+
+In terms of linking, they are managed very similarly to hyperlinks. However,
+they are placed on their own line; are preceded by an exclamation point; and
+both the label and destination have a specific naming convention:
+
+```no-highlight
+![alt attribute][img__webui_descriptor]
+```
+
+The bottom of the file will look like:
+
+```no-highlight
+[img__webui_descriptor]: ../media/filename__webui_descriptor.png
+```
+
+Explanation:
+
+- `filename`: name of file containing the image (omit the extension '.md')
+- `webui`: version of software corresponding to the image of the web UI
+- `alt attribute`: text that is shown in place of the image if the latter
+  cannot be displayed for some reason
+- `descriptor`: a short description of the image (e.g. 'enable-dhcp')
+
+For example:
+
+```no-highlight
+![enable dhcp][img__2.1_enable-dhcp]
+![enable fire alarm][img__enable-fire-alarm]
+
+...
+
+[img__2.1_enable-dhcp]: ../media/installconfig-networking-dhcp__2.1_enable-dhcp.png
```

Comment:
This is inherited directly from MAAS. It's arduous, yes, but something like this may work best if we ever automate image generation. I don't have a big problem with it tbh, and we should attempt to agree on a standard. I suggest we arrange another meeting.